test/js/bun/console/console-symbol-iterator-run.ts

```diff
@@ -0,0 +1,33 @@
+async function readInputUsingForOf() {
+  const items = [];
+  for await (const line of console) {
+    items.push(`FIRST${line}`);
+    break;
+  }
+  for await (const line of console) {
+    if (line == "break") {
+      break;
+    }
+    items.push(line);
+  }
+  return items;
+}
+
+async function readInputUsingSymbolAsyncIterator() {
+  const items = [];
+  const firstLine = (await console[Symbol.asyncIterator]().next()).value;
```

Comment:
btw is this actually an expected usage? I am debugging and I saw that it's not returning so the reader will never be released. So I figured we might first decide whether it's expected to work.

with this diff:

```diff
diff --git a/test/js/bun/console/console-symbol-iterator-run.ts b/test/js/bun/console/console-symbol-iterator-run.ts
index f1c5efbfa..4392831ec 100644
--- a/test/js/bun/console/console-symbol-iterator-run.ts
+++ b/test/js/bun/console/console-symbol-iterator-run.ts
@@ -15,7 +15,9 @@ async function readInputUsingForOf() {
 
 async function readInputUsingSymbolAsyncIterator() {
   const items = [];
-  const firstLine = (await console[Symbol.asyncIterator]().next()).value;
+  const it = console[Symbol.asyncIterator]();
+  const firstLine = (await it.next()).value;
+  await it.return?.();
   items.push(`FIRST${firstLine}`);
   for await (const line of console) {
     if (line == "break") {
```

It will be working as expected.

Reply:
should it work if the user never releases the generator?